web/src/graph-viewer.ts

```diff
@@ -94,61 +93,85 @@ export class GraphViewer extends LitElement {
   }
 
   render() {
+    if (this.queryResult == null) return null;
+
+    if (this._errorMsg != null)
+      return html`<div id="error">${this._errorMsg}</div>`;
+
     return html`
       <sl-tab-group>
-        <sl-tab slot="nav" panel="${TABS.graph}">Graph</sl-tab>
-        <sl-tab slot="nav" panel="${TABS.raw}">Text</sl-tab>
+        <sl-tab
+          slot="nav"
+          panel="${TABS.graph}"
+          ?active=${this._currentTab === TABS.graph}
+          >Graph</sl-tab
+        >
+        <sl-tab
+          slot="nav"
+          panel="${TABS.raw}"
+          ?active=${this._currentTab === TABS.raw}
+          >Text</sl-tab
+        >
       </sl-tab-group>
 
       ${choose<TabType, TemplateResult>(this._currentTab, [
         [
           TABS.graph,
           () => html`
             <div id="cy-container">
-              ${this._error
-                ? html`<div id="error">Error fetching graph data :(</div>`
-                : html`<div id="cy"></div>`}
+              <div id="cy"></div>
             </div>
           `,
         ],
         [
           TABS.raw,
-          () => html` <div id="raw-data">${this.#graphData.raw}</div> `,
+          () => html`<div id="raw-data">${this.queryResult.data.raw}</div>`,
         ],
       ])}
     `;
   }
 
   async updated(changedProperties: Map<string, any>) {
     if (
-      this._error === false &&
+      this._errorMsg != null &&
       changedProperties.has("_currentTab") &&
       this._currentTab === TABS.graph
     ) {
       const cy = await this._cy;
-      renderCyGraph(this.#graphData, cy);
+      renderCyGraph(this.queryResult, cy);
     }
 
     if (this.queryResult != null && changedProperties.has("queryResult")) {
       // Server error
       if (this.queryResult.error === true) {
-        this._error = true;
+        this._errorMsg = "Error fetching graph data :(";
+        return;
+      }
+      // Empty results
+      if (emptyQueryResult(this.queryResult.data)) {
+        this._errorMsg = "No results :(";
         return;
       }
 
       try {
-        this._error = false; // Remove old errors
-        this.#graphData = this.queryResult.data;
+        this._errorMsg = null; // Remove old errors
         const cy = await this._cy;
-        renderCyGraph(this.#graphData, cy);
+        if (this._currentTab === TABS.graph)
+          renderCyGraph(this.queryResult, cy);
       } catch {
-        // Server response is OK, but we can't render it
-        this._error = true;
+        this._errorMsg = "Error";
```

Comment:
Could we add in a bit more? What should we say to the user? What should they do? Contact the administrator? How can they do that?

Reply:
That error should never happen, I added a message to contact us with GitHub, what do you think?
`Something went wrong :(
 Please try again later, if the error persists contact us on GitHub`